address.js

```diff
@@ -125,24 +124,33 @@ proto.extract = function(fieldName, regexes) {
   }
 
   // iterate over the unit regexes and test them against the various parts
-  for (rgxIdx = 0; rgxIdx < regexes.length; rgxIdx++) {
-    for (ii = this.parts.length; ii >= 0; ii-- ) {
+  for (ii = this.parts.length-1; ii >= 0; ii--) {
```

Comment:
While you are in the neighbourhood... I don't think there is any advantage in the decrementing value here (I was on a bit of a performance joyride at the time this was written with another library) so I think going back to the following is probably better for readability:

```js
for (ii = 0; ii < this.parts.length; ii++) {
}
```

In a subsequent PR I might look at replacing all those `var` statements with appropriate `const` and `let` equivalents also.  Probably time to bring it slowly into 2019 (without completely alienating older browsers).

Reply:
Matching tokens right-to-left works best for left-to-right languages.

It's the same technique I use for https://github.com/pelias/placeholder

If we changed this around then the tests would fail as abbreviations such as `Mt` in the "venue name" portion of the text would match on state, however those tokens are more likely to appear at the end of the input in English.

Reply:
Cool - sounds like you know what you are doing :)